crates/edr_eth/src/remote/client.rs

```diff
@@ -220,27 +220,39 @@ impl RpcClient {
         })
     }
 
-    fn parse_response_str<T: DeserializeOwned>(response: &str) -> Result<T, RpcClientError> {
-        serde_json::from_str(response).map_err(|error| RpcClientError::InvalidResponse {
-            response: response.to_string(),
-            expected_type: std::any::type_name::<T>(),
+    fn parse_response_str<T: DeserializeOwned>(
+        response: String,
+    ) -> Result<jsonrpc::Response<T>, RpcClientError> {
+        serde_json::from_str(&response).map_err(|error| RpcClientError::InvalidResponse {
+            response,
+            expected_type: std::any::type_name::<jsonrpc::Response<T>>(),
             error,
         })
     }
 
-    fn extract_result<T: DeserializeOwned>(
+    async fn retry_on_sporadic_failure<T: DeserializeOwned>(
+        &self,
+        error: jsonrpc::Error,
         request: SerializedRequest,
-        response: String,
     ) -> Result<T, RpcClientError> {
-        let response: jsonrpc::Response<T> = Self::parse_response_str(&response)?;
+        fn is_missing_trie_node_error(error: &jsonrpc::Error) -> bool {
+            error.code == -32000 && error.message.contains("missing trie node")
+        }
```

Comment:
Our of curiosity, why is this a function and not a variable? That is, why not do this instead:

```rust
        let is_missing_trie_node_error =
            error.code == -32000 && error.message.contains("missing trie node");

        let result = if is_missing_trie_node_error {
```

Is using a function more idiomatic in Rust for some reason?

Reply:
There is no particular reason. I actually don't know why I didn't do it your way; I like it better 😁 

Changed in https://github.com/NomicFoundation/edr/pull/410/commits/0519bc68a5dbed5ac60ba299d30712f5cb63e233